### PR TITLE
IMessageService enhancements to support the 'AllowDismiss' property

### PR DESCRIPTION
**Pull request by StevenRasmussen**
# Pull Request

## 📖 Description

Building off of the recently released version which added the ability to prevent dismissing of the message bar notification in the `MessageBar` component, this PR adds overloads to the `IMessageService` which can be used for adding message bar components to a `MessageBarProvider` instance.

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fluentui-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [x] I have modified an existing component
- [x] I have validate [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

File: src/Core/Components/MessageBar/Services/IMessageService.cs [hunk 1]
```diff
@@ -16,25 +16,43 @@ public interface IMessageService : IDisposable
     Message ShowMessageBar(Action<MessageOptions> options);
 
     /// <summary />
-    Message ShowMessageBar(string message);
+    Message ShowMessageBar(string title);
 
     /// <summary />
-    Message ShowMessageBar(string message, MessageIntent intent);
+    Message ShowMessageBar(string title, bool allowDismiss);
```
Comment (dvoituron) [inline src/Core/Components/MessageBar/Services/IMessageService.cs:22]: I don't think we'd like to make it so easy to remove the close icon.

In FluentUI, it's always preferable for the users to be able to close a message if they wish.

However, adding the option in MessageOptions (only) is still a good idea, I think.

File: src/Core/Components/MessageBar/Services/IMessageService.cs [hunk 2]
```diff
@@ -16,25 +16,43 @@ public interface IMessageService : IDisposable
...
 
     /// <summary />
-    Message ShowMessageBar(string message, MessageIntent intent, string section);
+    Message ShowMessageBar(string title, MessageIntent intent);
```
Comment (dvoituron) [inline src/Core/Components/MessageBar/Services/IMessageService.cs:25]: Why rename `message` to `title`?

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: src/Core/Components/MessageBar/Services/IMessageService.cs (hunk 1 above)
Comment (StevenRasmussen) [reply src/Core/Components/MessageBar/Services/IMessageService.cs:22]: Ok.. I will remove the overloads.

File: src/Core/Components/MessageBar/Services/IMessageService.cs (hunk 2 above)
Comment (StevenRasmussen) [reply src/Core/Components/MessageBar/Services/IMessageService.cs:25]: The implementation (`MessageService`) had the parameter named `title` so I was merely updating it for consistency.  If the better name is `message` then I will update everything to be `message.

Comment (StevenRasmussen) [reply src/Core/Components/MessageBar/Services/IMessageService.cs:25]: It appears that the underlying `Options` object has a `Title` property on it that the `message` parameter is being mapped to... perhaps that's why the parameter was named `title` on the implementation side.  Either way, I think the interface and the implementation should have the same name.  What would you like that to be?

Comment (vnbaaij) [reply src/Core/Components/MessageBar/Services/IMessageService.cs:25]: I want to follow the design guidelines as much as possible:
image

We are already streching that with th option to not have the close button

Going by what we are doing in the example 'Add on top',
image

I would say that that is showing the Title (number 3 from the design image). So I think `MessageService.cs` is correct and `IMessageService.cs` needs to be changed. Agree?

Comment (dvoituron) [reply src/Core/Components/MessageBar/Services/IMessageService.cs:25]: Yes. Correct

File: src/Core/Components/MessageBar/Services/IMessageService.cs (hunk 1 above)
Comment (StevenRasmussen) [reply src/Core/Components/MessageBar/Services/IMessageService.cs]: Removed

File: src/Core/Components/MessageBar/Services/IMessageService.cs (hunk 2 above)
Comment (StevenRasmussen) [reply src/Core/Components/MessageBar/Services/IMessageService.cs:22]: Both the `IMessageService` and the `MessageService` implementation use `title` now.

File: src/Core/Components/MessageBar/Services/IMessageService.cs (hunk 1 above)
Comment (dvoituron) [reply src/Core/Components/MessageBar/Services/IMessageService.cs]: I think the arguments are still in the `MessageService` class, whereas they are no longer in `IMessageService`.

Comment (StevenRasmussen) [reply src/Core/Components/MessageBar/Services/IMessageService.cs]: Good catch!  Removed.

**Review by dvoituron — APPROVED** [review]

**Pull request merged**